internal/app/machined/pkg/controllers/secrets/kubernetes_cert_sans_test.go

```diff
@@ -137,6 +140,54 @@ func (suite *KubernetesCertSANsSuite) TestReconcile() {
 			},
 		),
 	)
+
+	_, err = suite.state.UpdateWithConflicts(suite.ctx, rootSecrets.Metadata(), func(r resource.Resource) error {
+		r.(*secrets.KubernetesRoot).TypedSpec().Endpoint, err = url.Parse("https://some.other.url:6443/")
+
+		return err
+	})
+	suite.Require().NoError(err)
+
+	suite.Assert().NoError(
+		retry.Constant(10*time.Second, retry.WithUnits(100*time.Millisecond)).Retry(
+			func() error {
+				var certSANs resource.Resource
+
+				certSANs, err = suite.state.Get(
+					suite.ctx,
+					resource.NewMetadata(
+						secrets.NamespaceName,
+						secrets.CertSANType,
+						secrets.CertSANKubernetesID,
+						resource.VersionUndefined,
+					),
+				)
+				if err != nil {
+					return err
+				}
+
+				spec := certSANs.(*secrets.CertSAN).TypedSpec()
+
+				expectedDNSNames := []string{
+					"example.com",
+					"foo",
+					"foo.example.com",
+					"kubernetes",
+					"kubernetes.default",
+					"kubernetes.default.svc",
+					"kubernetes.default.svc.cluster.remote",
+					"localhost",
+					"some.other.url",
+				}
+
+				if !reflect.DeepEqual(spec.DNSNames, expectedDNSNames) {
```

Comment:
is this checking if it's reset?

Reply:
it's checking that if we remove some name from the input list, it will disappear from the final generated list of cert SANs.

we mutate the input and verify that output is correct

that's a regression test - if we remove .Reset() call, the test will fail